src/server/.tests/passport/strategyUtils.spec.js

```diff
@@ -0,0 +1,168 @@
+const strategyUtils = require('../../passport/strategyUtils')
+const responseManager = require('../../routes/controllers/utilities/responseManager')
+const assert = require('chai').assert
+const sinon = require('sinon')
+const httpMocks = require('node-mocks-http')
+const EventEmitter = require('events').EventEmitter
+
+let suite = {}
+describe('StrategyUtils', () => {
+	beforeEach(() => {
+		suite = {}
+	})
+
+	describe('#strategyHandlers', () => {
+		beforeEach(() => {
+			suite.requestMock = createRequestMock()
+			suite.nextSpy = sinon.spy()
+			suite.responseMock = httpMocks.createResponse({
+				eventEmitter: EventEmitter
+			})
+			suite.strategyHandlersMock = strategyUtils.strategyHandlers(suite.requestMock, suite.responseMock, suite.nextSpy)
+		})
+		describe('#serializeHandler', () => {
+			it('should call next when no error is given', () => {
+				//given
+				const error = null
+
+				//when
+				suite.strategyHandlersMock.serializeHandler(error)
+
+				//then
+				assert.isTrue(suite.nextSpy.calledOnce)
+			})
+
+			it('should respond with error status code and description', () => {
+				//given
+				const error = {
+					code: 401,
+					description: 'UNAUTHORIZED'
+				}
+
+				//when
+				suite.strategyHandlersMock.serializeHandler(error)
+
+				//then
+				const statusCode = suite.responseMock._getStatusCode()
+				const payload = suite.responseMock._getData()
+				const expectedStatusCode = error.code
+				const expectedPayload = JSON.stringify({
+					description: error.description
+				})
+
+				assert.strictEqual(statusCode, expectedStatusCode)
+				assert.strictEqual(payload, expectedPayload)
+			})
+		})
+
+		describe('#loginStrategyHandler', () => {
+			it('should respond with error status code and description, when error is given', () => {
+				//given
+				const error = {
+					code: 401,
+					description: 'UNAUTHORIZED'
+				}
+				const userInstance = {}
+				const info = null
+
+				//when
+				suite.strategyHandlersMock.loginStrategyHandler(error, userInstance, info)
+
+				//then
+				const statusCode = suite.responseMock._getStatusCode()
+				const payload = suite.responseMock._getData()
+				const expectedStatusCode = error.code
+				const expectedPayload = JSON.stringify({
+					description: error.description
+				})
+
+				assert.strictEqual(statusCode, expectedStatusCode)
+				assert.strictEqual(payload, expectedPayload)
+
+			})
+
+			it('should respond with {description: UNATHORIZED , code: 401}, when userInstance is null', () => {
```

Comment:
Just `should respond with UNAUTHORIZED when...`

Reply:
Shouldn't we respond with BAD REQUEST?

Reply:
ok